### ISSUE TITLE
Array support for Data

### ISSUE THREAD

**Issue by derekdreery** [xi-editor/druid#884]
would be nice to have

```rust
impl<T: Data, const Size: usize> Data for [T; Size]
```

or the equivalent macro for pre-const generics, like

```rust
macro_rules! impl_Data {
    () => {},
    ($this:expr $($rest:tt)*) => {
        impl<T: Data> Data for [T; $this] {
            fn same(&self, other: &Self) -> bool {
                self.iter().zip(other.iter()).all(|(a, b)| a.same(b))
            }
        }
        impl_Data($(rest)*);
    }
}

impl_Data { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 32, 64, 128 } // etc

**Comment by derekdreery** [comment]
The macro probably has errors in it but that's the gist.

**Comment by cmyr** [comment]
would happily accept this as a PR for arrays up to some size (for now), say 32?

**Comment by derekdreery** [comment]
Fixed in #888
